tests/test_config.py

```diff
@@ -159,7 +159,7 @@ def test_ssl_config_combined(tls_certificate_pem_path):
     assert config.is_ssl is True
 
 
-def asgi2_app(scope):
+def asgi2_app(scope):  # pragma: no cover
```

Comment:
this should not be skipped imho, we have code in config.py that makes the switch from asgi2 to asgi3 : https://github.com/encode/uvicorn/blob/d81b96ee0590df0b5111376ba38637645385a684/uvicorn/config.py#L331-L339

Reply:
You're right. But the body of that function is not executed, because the `__call__` on the `ASGI2Middleware` is never reached. 

https://github.com/encode/uvicorn/blob/e6028a6b787f13692d04b4e5aca67a6319b8360e/uvicorn/middleware/asgi2.py#L5-L7

Reply:
lgtm thanks !